arkouda/segarray.py

```diff
@@ -393,7 +403,14 @@ def get_ngrams(self, n, return_origins=True):
                 valid &= notsegstart[i:end]
         ngrams = [char[valid] for char in ngrams]
         if return_origins:
-            origin_indices = self.grouping.broadcast(arange(self.size), permute=True)[:valid.size][valid]
+            # set the proper indexes for broadcasting. Needed to alot for empty segments
+            seg_idx = zeros(self.non_empty_count, dtype=akint64)
+            idx = 0
+            for i in range(self.size):
+                if self.non_empty[i]:
+                    seg_idx[idx] = i
+                    idx += 1
```

Comment:
I think this can be accomplished with this
```python
seg_idx = arange(self.size)[self.non_empty]
```

Reply:
Updated to your suggestion.